Sources/BigInt/BigInt.swift

```diff
@@ -330,37 +330,29 @@ extension BigInt: SignedNumeric {
 extension BigInt: BinaryInteger {
 
   public init?<T>(exactly source: T) where T: BinaryFloatingPoint {
-    if (source.isNaN || source.isInfinite) ||
-      (source.rounded(.towardZero) != source) {
+    guard source.isFinite, source == source.rounded(.towardZero) else {
       return nil
     }
-
     self.init(source)
   }
 
   public init<T>(_ source: T) where T: BinaryFloatingPoint {
-    precondition(
-      !(source.isNaN || source.isInfinite),
-      "\(type(of: source)) value cannot be converted to BigInt because it is either infinite or NaN"
-    )
-
-    let isNegative = source < 0.0
-    var float = isNegative ? -source : source
+    precondition(source.isFinite, "BigInt(\(source)) failed")
```

Comment:
The verbiage I used was copied directly from what `Int` et al say in this case, so should probably be what we go with.

Reply:
Done. I've kept the `source.isFinite` condition, because it matches what the [standard library][] uses.

[standard library]: <https://github.com/apple/swift/blob/958326151c63ab3d8892dddd5b960125e3b97a37/stdlib/public/core/IntegerTypes.swift.gyb#L1156-L1157>

Reply:
Right, that part was fine and an obvious improvement, it was just the error message I thought should also match the stdlib ;)